packages/react-router/index.ts

```diff
@@ -25,32 +39,49 @@ import type {
   PathPattern,
   RedirectFunction,
   RelativeRoutingType,
+  RevalidationState,
   Router as RemixRouter,
+  RouterState,
+  RouterSubscriber,
   FutureConfig as RouterFutureConfig,
   ShouldRevalidateFunction,
   ShouldRevalidateFunctionArgs,
+  StaticHandlerContext,
   To,
   UIMatch,
+  UNSAFE_RouteManifest,
   unstable_HandlerResult,
-} from "@remix-run/router";
+} from "./router";
 import {
   AbortedDeferredError,
   Action as NavigationType,
+  createBrowserHistory,
+  createHashHistory,
```

Comment:
This feels weird - eventually maybe these implementations should just live in `react-router-dom`?  But that feels weird too - maybe another vote for keeping a standalone router package to colocate histories and let RR and RR-dom can both import from there...